openhands/runtime/utils/request.py

```diff
@@ -58,5 +22,12 @@ def send_request(
     **kwargs: Any,
 ) -> requests.Response:
     response = session.request(method, url, **kwargs)
-    response.raise_for_status()
+    try:
+        response.raise_for_status()
+    except requests.HTTPError as e:
+        try:
+            _json = response.json()
+        except requests.JSONDecodeError:
+            raise e
+        raise RequestError(e, details=_json.get('detail')) from e
```

Comment:
I think we specifically handle `HTTPError` in many places--this will start swallowing them

Reply:
`RequestError` is actually inherit from requests.HTTPError with an overloaded `__str__` method, which should hurt? But i could do a global replacement of requests.HTTPError for this though

Reply:
done in most recent commit